MatrixSDK/Data/MXRoomLastMessage.m

```diff
@@ -0,0 +1,305 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "MXRoomLastMessage.h"
+#import "MXEvent.h"
+
+#import <Security/Security.h>
+#import <CommonCrypto/CommonCryptor.h>
+
+NSString *const kCodingKeyEventId = @"eventId";
+NSString *const kCodingKeyOriginServerTs = @"originServerTs";
+NSString *const kCodingKeyIsEncrypted = @"isEncrypted";
+NSString *const kCodingKeySender = @"sender";
+NSString *const kCodingKeyData = @"data";
+NSString *const kCodingKeyEncryptedData = @"encryptedData";
+NSString *const kCodingKeyText = @"text";
+NSString *const kCodingKeyAttributedText = @"attributedText";
+NSString *const kCodingKeyOthers = @"others";
+
+@interface MXRoomLastMessage ()
+
+@property (nonatomic, copy, readwrite) NSString *eventId;
+
+@property (nonatomic, assign, readwrite) uint64_t originServerTs;
+
+@end
+
+@implementation MXRoomLastMessage
+
+- (instancetype)initWithEvent:(MXEvent *)event
+{
+    if (self = [super init])
+    {
+        _eventId = event.eventId;
+        _originServerTs = event.originServerTs;
+        _isEncrypted = event.isEncrypted;
+        _sender = event.sender;
+        _others = [NSMutableDictionary new];
```

Comment:
We can keep it nil. The property is nullable.

Reply:
Updated, some usage will be updated as well.